File: app/src/main/java/com/hbjpro/androidvoiceassistant/Model/ModelSpeech.java
package com.hbjpro.androidvoiceassistant.Model;

import android.content.Intent;

import com.hbjpro.androidvoiceassistant.Command.CommandManager;
import com.hbjpro.androidvoiceassistant.Interface.OnOrderResult;
import com.hbjpro.androidvoiceassistant.Interface.OnSpeechManager;
import com.hbjpro.androidvoiceassistant.MainActivity;
import com.hbjpro.androidvoiceassistant.Speech.SpeechManager;
import com.hbjpro.androidvoiceassistant.Speech.SpeechResult;
import com.hbjpro.androidvoiceassistant.Tools.Tools;

public class ModelSpeech implements OnSpeechManager, OnOrderResult{
    private MainActivity main;

    public ModelSpeech(MainActivity main){
        this.main = main;
    }

    /* --- Speech Manager --- */
    public void startSpeech(Tools.LanguageCode languageCode){
        SpeechManager speechManager = new SpeechManager(languageCode, this);
        speechManager.speech(main);
    }

    /* --- Order Manager --- */

    /* --- OnSpeechManager --- */
    @Override
    public void onSpeechManagerFinish(SpeechResult speechResult) {
        CommandManager commandManager = new CommandManager(this);
        commandManager.executeCommand(speechResult);
    }

    /* --- OnOrderResult --- */
    @Override
    public void onGetLaunchIntentSuccess(Intent launchIntent) {
        main.onOpenAppSuccess(launchIntent);
    }

    @Override
    public void onGetLaunchIntentError() {

    }
}

File: app/src/main/java/com/hbjpro/androidvoiceassistant/Model/ModelSpeech.kt
package com.hbjpro.androidvoiceassistant.Model

import android.content.Intent
import com.hbjpro.androidvoiceassistant.Command.CommandManager
import com.hbjpro.androidvoiceassistant.Interface.OnOrderResult
import com.hbjpro.androidvoiceassistant.Interface.OnSpeechManager
import com.hbjpro.androidvoiceassistant.MainActivity
import com.hbjpro.androidvoiceassistant.Speech.SpeechManager
import com.hbjpro.androidvoiceassistant.Speech.SpeechResult
import com.hbjpro.androidvoiceassistant.Tools.Tools

class ModelSpeech(val main: MainActivity) : OnSpeechManager, OnOrderResult {

    /* --- Speech Manager --- */
    fun startSpeech(languageCode: Tools.LanguageCode){
        var speechManager = SpeechManager(languageCode, this)
        speechManager.speech(main)
    }

    /* --- OnSpeechManager --- */
    override fun onSpeechManagerFinish(speechResult: SpeechResult?) {
        var commandManager = CommandManager(this)
        commandManager.executeCommand(speechResult)
    }

    /* --- OnOrderResult --- */
    override fun onGetLaunchIntentSuccess(launchIntent: Intent?) {
        main.onOpenAppSuccess(launchIntent)
    }

    override fun onGetLaunchIntentError() {

    }
}